go/coordinator/internal/memberlist_manager/memberlist_store.go

```diff
@@ -0,0 +1,88 @@
+package memberlist_manager
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+type IMemberlistStore interface {
+	GetMemberlist() (*Memberlist, error)
+	UpdateMemberlist(memberlist *Memberlist) error
+}
+
+type Memberlist []string
+
+type CRMemberlistStore struct {
+	dynamicClient            dynamic.Interface
+	coordinatorNamespace     string
+	memberlistCustomResource string
+}
+
+func NewCRMemberlistStore(dynamicClient dynamic.Interface, coordinatorNamespace string, memberlistCustomResource string) *CRMemberlistStore {
+	return &CRMemberlistStore{
+		dynamicClient:            dynamicClient,
+		coordinatorNamespace:     coordinatorNamespace,
+		memberlistCustomResource: memberlistCustomResource,
+	}
+}
+
+func (s *CRMemberlistStore) GetMemberlist() (*Memberlist, error) {
+	gvr := getGvr()
+	unstrucuted, err := s.dynamicClient.Resource(gvr).Namespace("chroma").Get(context.TODO(), "worker-memberlist", metav1.GetOptions{}) //.Namespace(m.coordinator_namespace).Get(context.TODO(), m.memberlist_custom_resource, metav1.GetOptions{})
```

Comment:
TODO: investigate how to use context properly (pass into struct?)

Reply:
https://go.dev/blog/context-and-structs#:~:text=Contexts%20should%20not%20be%20stored,store%20it%20in%20another%20type.